ilastik/applets/dataSelection/datasetInfoEditorWidget.py

```diff
@@ -249,31 +251,35 @@ def _applyTempOpSettingsRealOp(self):
             else:
                 originalInfos[laneIndex] = None
 
+        ret = None
         try:
             for laneIndex, op in list(self.tempOps.items()):
-                info = copy.copy( op.Dataset.value )
+                info = copy.copy(op.Dataset.value)
                 realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
-                realSlot.setValue( info )
-            return True
+                realSlot.setValue(info)
+        except DatasetConstraintError as ex:
+            ret = ex
+            # Try to revert everything back to the previous state
+            try:
+                for laneIndex, info in list(originalInfos.items()):
+                    realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
+                    if realSlot is not None:
+                        realSlot.setValue(info)
+            except Exception:
```

Comment:
this one is of course never really nice, or does this get caught then by the outher `except`?